src/main/java/com/mcmoddev/mmdbot/modules/commands/server/quotes/CmdGetQuote.java

```diff
@@ -72,16 +72,14 @@ protected void execute(final CommandEvent event) {
             }
         }
 
-        // Get a random quote.
-        int limit = QuoteList.getQuoteSlot() - 1;
-        int index = new Random().nextInt(limit);
-        Quote fetched = QuoteList.getQuote(index);
-
-        // Check if the quote exists.
-        if (fetched == null) {
-            // Send the standard message
-            channel.sendMessageEmbeds(QuoteList.getQuoteNotPresent()).queue();
-            return;
+        Quote fetched = null;
+        // Find a valid quote.
+        while (fetched == null) {
+            // Get a random quote.
+            int limit = QuoteList.getQuoteSlot();
+            int index = new Random().nextInt(limit);
+            // If the next quote is null, we loop back and try again.
+            fetched = QuoteList.getQuote(index);
```

Comment:
Minor things:

- The `limit` could be moved outside the loop (unless we want to anticipate a situation where the quotes list increases while this is iterating over it, which is unlikely).
- The `Random` can be stored instead of creating a new one on each loop iteration.
- You could change the loop into a `do { ... } while (fetched == null);`, so the first time the condition is checked, it may or may not pass the check, instead of currently where the first time, it's always `null`.
- This may possibly infinitely loop if there are 0 quotes stored (unlikely, but the possibility exists when the bot is freshly set up).